src/zjs_gpio.c

```diff
@@ -13,6 +15,74 @@
 #include "zjs_callbacks.h"
 #include "zjs_promise.h"
 
+#if defined(ZJS_LINUX_BUILD) || defined(QEMU_BUILD)
+#define ZJS_GPIO_MOCK
+#endif
+
+#ifdef ZJS_GPIO_MOCK
+#ifdef ZJS_LINUX_BUILD
+// provide Zephyr dependencies
+struct device;
+struct gpio_callback;
+typedef void (*gpio_callback_handler_t)(struct device *port,
```

Comment:
Good idea, this actually let me remove the evil casting between struct device * and jerry_value_t.